lib/global-admin/addon/clusters/new/cloud/controller.js

```diff
@@ -38,6 +39,22 @@ export default Controller.extend(NewOrEdit, {
     ];
   }),
 
+  validate() {
```

Comment:
Can you remove the commit for this PR?  The server-side should be doing unique checks and returning 422 on POST if it's non-unique.  Checking in the client is error prone (race conditions, or a cluster you don't have access to see exists).  Just make sure if there's an error returned from the API that it's surfaced to the user (I think that's the actual problem in rancher/rancher#10855).

(also as of a day or two ago cluster id is not actually the name anymore so the name does not really have to be unique)